### PR TITLE
Added FileDetailLayout

### PR DESCRIPTION
**Pull request by adlius**
-   Ticket: []
-   Feature flag: n/a

## Purpose

Added FileDetailLayout component and make use of such component.

### CODE REVIEW

File: app/guid-file/controller.ts [hunk 1]
```diff
@@ -1,16 +1,19 @@
 /* eslint-disable no-console */
 import Controller from '@ember/controller';
 import { action } from '@ember/object';
+// import { action } from '@ember/object';
```
Comment (brianjgeiger) [inline app/guid-file/controller.ts:4]: Does this need to be here?

File: app/guid-file/-components/file-detail-layout/component.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+// import { action } from '@ember/object';
+// import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import Component from '@glimmer/component';
+// import { tracked } from '@glimmer/tracking';
+// import { taskFor } from 'ember-concurrency-ts';
+// import Media from 'ember-responsive';
```
Comment (brianjgeiger) [inline app/guid-file/-components/file-detail-layout/component.ts:7]: Do the commented out imports need to stay?

File: app/guid-file/-components/file-detail-layout/component.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+interface Args {
+    rightColumnClosed: boolean;
+}
+
+export default class FileDetailLayout extends Component<Args> {
+    @action
+    noop() {
```
Comment (brianjgeiger) [inline app/guid-file/-components/file-detail-layout/component.ts:15]: Is this component file doing anything, or could this component be template-only?

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
Looks good. My only complaint is that the menu is now in a place where it's unusable. Could you scoot that back over to the right-side of the section to keep from losing functionality?

File: app/guid-file/-components/file-detail-layout/component.ts (hunk 3 above)
Comment (adlius) [reply app/guid-file/-components/file-detail-layout/component.ts:15]: Just a boilerplate. I dunno whether there will be anything new added to the layout component so I left it there. Could be removed.

Comment (brianjgeiger) [reply app/guid-file/-components/file-detail-layout/component.ts:15]: Might as well take it out until we need it, I'm thinking.

File: app/guid-file/-components/file-detail-layout/styles.scss [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    flex-direction: row;
+    flex-grow: 1;
+}
+
+.MainColumn {
+    flex-grow: 3;
+}
+
+.RightColumn {
+    flex-grow: 1;
```
Comment (futa-ikeda) [inline app/guid-file/-components/file-detail-layout/styles.scss:12]: Right now, for smaller desktop sizes this column gets a bit squished and leads to some text wrapping, so adding a `min-width:300px` or something like that should prevent some of those issues as we develop the tags and revisions widgets.

File: app/guid-file/template.hbs [hunk 5]
```diff
@@ -1,235 +1,188 @@
...
-        </div>
-    </div>
-</div>
+        {{/if}}
+    </:right>
+    <:rightButtons>
+        {{#if this.media.isMobile}}
+            <BsButton
+                data-test-versions-button
+                data-analytics-name='Versions button'
```
Comment (futa-ikeda) [inline app/guid-file/template.hbs:161]: A couple copy-pastas

**Review by futa-ikeda — CHANGES_REQUESTED** [review]
Looks great! Just a couple things that caught my eye

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
There is a width at which the icons go side-by-side, but the other changes don't happen to the tabs. It looks like this:

Screen Shot 2022-03-07 at 3 48 50 PM

Other than that, it seems to be working quite well.

**Comment by brianjgeiger** [comment]
Oh, also: clicking on tabs items twice in mobile view shouldn't change the view; when in mobile view, ideally these should work just like normal tabs. It's especially odd when you click on the file tab twice and it switches you to the revisions tab.

**Review by brianjgeiger — APPROVED** [review]

**Review by futa-ikeda — APPROVED** [review]

**Pull request merged**